### PR TITLE
F: V1 Deprecate /nearby address and bing map support 

### PR DESCRIPTION
**Pull request by taylorrassmannlibertyits**
https://vajira.max.gov/browse/API-2095
Refactored for v0 to v1 update removing nearby searches on address for v1.

### CODE REVIEW

**Review by jnguy-va — APPROVED** [review]
v0 and v1 looks good on my end.

**Review by xenkidu-va — APPROVED** [review]
LGTM

File: facilities/src/main/java/gov/va/api/lighthouse/facilities/v1/NearbyControllerV1.java [hunk 1]
```diff
@@ -0,0 +1,273 @@
...
+@Slf4j
+public class NearbyControllerV1 {
+
+  private static final Set<Integer> DRIVE_TIME_VALUES = Set.of(10, 20, 30, 40, 50, 60, 70, 80, 90);
+
+  private final FacilityRepository facilityRepository;
+
+  private final DriveTimeBandRepository driveTimeBandRepository;
+
+  private final NearbyControllerV1.DeprecatedPssgDriveTimeBandSupport
```
Comment (as-va) [inline facilities/src/main/java/gov/va/api/lighthouse/facilities/v1/NearbyControllerV1.java:57]: This file is full of `NearbyControllerV1.` prefixes on references to classes defined in this file. These references do not need them.
Remove `NearbyControllerV1.`

File: facilities/src/main/java/gov/va/api/lighthouse/facilities/v1/NearbyControllerV1.java [hunk 2]
```diff
@@ -0,0 +1,273 @@
...
+      deprecatedPssgDriveTimeBandSupport =
+          new NearbyControllerV1.DeprecatedPssgDriveTimeBandSupport();
```
Comment (as-va) [inline facilities/src/main/java/gov/va/api/lighthouse/facilities/v1/NearbyControllerV1.java:59]: Remove `NearbyControllerV1.`

File: facilities/src/main/java/gov/va/api/lighthouse/facilities/v1/NearbyControllerV1.java [hunk 3]
```diff
@@ -0,0 +1,273 @@
...
+            entities.size());
+        return Optional.of(entity);
+      }
+    }
+    log.info("No matches found in {} options", entities.size());
+
+    return Optional.empty();
+  }
+
+  private String getMonthYearFromBandIds(List<NearbyControllerV1.NearbyId> ids) {
```
Comment (as-va) [inline facilities/src/main/java/gov/va/api/lighthouse/facilities/v1/NearbyControllerV1.java:99]: Remove `NearbyControllerV1.`

File: facilities/src/main/java/gov/va/api/lighthouse/facilities/v1/NearbyControllerV1.java [hunk 4]
```diff
@@ -0,0 +1,273 @@
...
+                  entry.getValue().stream()
+                      .sorted(Comparator.comparingInt(left -> left.id().fromMinutes()))
+                      .collect(toList());
+              return firstIntersection(point, sortedEntities).orElse(null);
+            })
+        .filter(Objects::nonNull)
+        .collect(toMap(b -> b.id().stationNumber(), Function.identity()));
+  }
+
+  private NearbyResponse.Nearby nearbyFacility(@NonNull NearbyControllerV1.NearbyId entity) {
```
Comment (as-va) [inline facilities/src/main/java/gov/va/api/lighthouse/facilities/v1/NearbyControllerV1.java:138]: Remove `NearbyControllerV1.`

File: facilities/src/main/java/gov/va/api/lighthouse/facilities/v1/NearbyControllerV1.java [hunk 5]
```diff
@@ -0,0 +1,273 @@
...
+        .attributes(
+            NearbyResponse.NearbyAttributes.builder()
+                .minTime(entity.bandId().fromMinutes())
+                .maxTime(entity.bandId().toMinutes())
+                .build())
+        .build();
+  }
+
+  @SneakyThrows
+  private List<NearbyControllerV1.NearbyId> nearbyIds(
```
Comment (as-va) [inline facilities/src/main/java/gov/va/api/lighthouse/facilities/v1/NearbyControllerV1.java:151]: Remove `NearbyControllerV1.`

File: facilities/src/main/java/gov/va/api/lighthouse/facilities/v1/NearbyControllerV1.java [hunk 6]
```diff
@@ -0,0 +1,273 @@
...
+  /** Nearby facilities by coordinates. */
+  @GetMapping(
+      produces = "application/json",
+      params = {"lat", "lng"})
+  NearbyResponse nearbyLatLong(
+      @RequestParam(value = "lat") BigDecimal latitude,
+      @RequestParam(value = "lng") BigDecimal longitude,
+      @RequestParam(value = "services[]", required = false) List<String> services,
+      @RequestParam(value = "drive_time", required = false) Integer maxDriveTime) {
+    List<NearbyControllerV1.NearbyId> ids = nearbyIds(longitude, latitude, services, maxDriveTime);
```
Comment (as-va) [inline facilities/src/main/java/gov/va/api/lighthouse/facilities/v1/NearbyControllerV1.java:202]: Remove `NearbyControllerV1.`

File: facilities/src/main/java/gov/va/api/lighthouse/facilities/WebExceptionHandlerV0.java [hunk 7]
```diff
@@ -5,6 +5,7 @@
 import com.google.common.collect.Iterables;
 import gov.va.api.health.autoconfig.configuration.JacksonConfig;
 import gov.va.api.lighthouse.facilities.api.v0.ApiError;
+import gov.va.api.lighthouse.facilities.v0.ApiExceptionsV0;
```
[the remark this answers is not in the document]
Comment (as-va) [reply facilities/src/main/java/gov/va/api/lighthouse/facilities/WebExceptionHandlerV0.java:8]: After discussion, moving this request to a new ticket.

**Pull request merged**